ddtrace/span.py

```diff
@@ -90,6 +92,7 @@ def __init__(
 
         # sampling
         self.sampled = True
+        self.priority = None
```

Comment:
Forgot to say, that probably all these "flags" should stay only in the `Context`. Am I missing something about sampled traces but discarded children spans?

Reply:
Overall I agree both should be moved there.

So far we had `sampled` in the span (old, from before the `Context` entity) but both should be moved.
I don't know how much pain that would be though (since we used to check `span.sampled` in various integrations) ; mayve it could wait a second PR?

Reply:
But yes, we have to fix this in the scope of this PR.
If we go with the `Context` approach to get/set the priority, let's not store it here. Plus whatever method we provide to get/set it ; it has to hit the right place.

(for now, the `span.set|get_sampling_priority` won't work with `context.get_context_attributes`).